File: app/src/main/java/me/devsaki/hentoid/activities/sources/PururinActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class PururinActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "pururin.to";
    private static final String[] GALLERY_FILTER = {"//pururin.to/gallery/"};

    Site getStartSite() {
        return Site.PURURIN;
    }

    @Override
    protected CustomWebViewClient createWebClient() {
        CustomWebViewClient client = new CustomWebViewClient(getStartSite(), GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/PururinActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

private const val DOMAIN_FILTER = "pururin.me"
const val PUR_GALLERY_PATTERN = "//pururin.me/browse/tags/collection/"
private val GALLERY_FILTER = arrayOf(PUR_GALLERY_PATTERN, "//pururin.me/gallery/")

class PururinActivity : BaseWebActivity() {

    override fun getStartSite(): Site {
        return Site.PURURIN
    }

    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        return client
    }
}